### PR TITLE
Work Around ASAN Deadlock in SpinQuic

### LINKED ISSUE #1933: ASAN Builds on Windows occasionally deadlock
It looks like theres currently a bug in ASAN that causes a deadlock. This manifests as a watchdog timeout in spinquic.

### PR DESCRIPTION
**Pull request by nibanks**
Hopefully fixes #1933 (or at least works around it).

### CODE REVIEW

File: scripts/spin.ps1 [hunk 1]
```diff
@@ -124,6 +124,8 @@ if ($CodeCoverage) {
 $SpinQuic = $null
 if ($IsWindows) {
     $SpinQuic = Join-Path $RootArtifactDir "spinquic.exe"
+    # Set ASAN option to prevent deadlock bug in ASAN RtlReAllocHeap.
+    $env:ASAN_OPTIONS='windows_hook_legacy_allocators=1'
```
Comment (thhous-msft) [inline scripts/spin.ps1:128]: This won't actually set the env for the process that gets started. In essence this does nothing.

Comment (thhous-msft) [reply scripts/spin.ps1:128]: What you instead need to do is in run-executable in the Start-Executable function after you create $pinfo is call `$pinfo.Environment.Add(key, value)`

**Comment by nibanks** [comment]
Thanks @thhouse-msft for fixing this! If you're good to go, let's approve/merge it!

**Comment by nibanks** [comment]
Work around causing even more issues. :(

**Pull request closed without merging**